src/libcore/fmt/mod.rs

```diff
@@ -1612,5 +1622,14 @@ impl<'b, T: ?Sized + Debug> Debug for RefMut<'b, T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.9.0")]
+impl<T: ?Sized + Debug> Debug for UnsafeCell<T> {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        f.debug_struct("UnsafeCell")
+            .field("value", &unsafe { &*self.get() })
```

Comment:
The unsafety is accessing the inner value to print, when another thread could be writing to it?